### PR TITLE
[test] Add PySpark check

### PR DESCRIPTION
**Pull request by rsarm**

### CODE REVIEW

**Comment by rsarm** [comment]
I changed the number of tasks to 3 to test the case of multiple worker nodes. Now we have 1 master and 2 workers.

**Review by teojgo — APPROVED** [review]

File: cscs-checks/apps/spark/spark_check.py [hunk 1]
```diff
@@ -10,26 +10,22 @@
 from reframe.core.backends import getlauncher
 
 
-@rfm.simple_test
-class SparkCheck(rfm.RunOnlyRegressionTest):
+class SparkBaseCheck(rfm.RunOnlyRegressionTest):
     def __init__(self):
-        self.descr = 'Simple calculation of pi with Spark'
         self.valid_systems = ['daint:gpu', 'daint:mc',
                               'dom:gpu', 'dom:mc']
```
Comment (vkarak) [inline cscs-checks/apps/spark/spark_check.py:16]: Can we add eiger here too?

File: cscs-checks/apps/spark/spark_check.py [hunk 2]
```diff
@@ -39,16 +35,44 @@ def prepare_run(self):
             exec_cores = 9
 
         self.variables = {
-            'SPARK_WORKER_CORES': '%s' % num_workers,
+            'SPARK_WORKER_CORES': f'{num_workers}',
```
Comment (vkarak) [inline cscs-checks/apps/spark/spark_check.py:38]:
```suggestion
            'SPARK_WORKER_CORES': str(num_workers),
```

File: cscs-checks/apps/spark/spark_check.py [hunk 3]
```diff
@@ -39,16 +35,44 @@ def prepare_run(self):
...
+    def prepare_run(self):
+        super().prepare_run()
+        self.executable_opts.extend([
+            '--class org.apache.spark.examples.SparkPi',
+            '$EBROOTSPARK/examples/jars/spark-examples*.jar 10000'
+        ])
+
+
+@rfm.simple_test
+class PySparkCheck(SparkBaseCheck):
```
Comment (vkarak) [inline cscs-checks/apps/spark/spark_check.py:70]: I think it's better if you use a single test and parametrize it. The two tests are almost identical.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: cscs-checks/apps/spark/spark_check.py (hunk 1 above)
Comment (rsarm) [reply cscs-checks/apps/spark/spark_check.py:16]: Spark is currently not on eiger's software stack.

File: cscs-checks/apps/spark/spark_check.py [hunk 4]
```diff
@@ -45,34 +50,14 @@ def prepare_run(self):
             f'--conf spark.executor.memory=15g',
             f'--master $SPARKURL'
         ]
+        if self.variant == 'spark':
+            self.executable_opts.extend([
+                '--class org.apache.spark.examples.SparkPi',
+                '$EBROOTSPARK/examples/jars/spark-examples*.jar 10000'
+            ])
```
Comment (vkarak) [inline cscs-checks/apps/spark/spark_check.py:57]:
```suggestion
            self.executable_opts += [
                '--class org.apache.spark.examples.SparkPi',
                '$EBROOTSPARK/examples/jars/spark-examples*.jar 10000'
            ]
```

**Review by vkarak — CHANGES_REQUESTED** [review]
I only have a minor comment, which I can fix.

**Review by vkarak — APPROVED** [review]

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**